vue/src/vueTextMask.js

```diff
@@ -0,0 +1,51 @@
+import createTextMaskInputElement from '../../core/src/createTextMaskInputElement.js'
+
+/*
+ * Vue Directive - Text Masking with Vanilla Text Mask
+ * https://github.com/text-mask/text-mask/tree/master/vanilla
+ *
+ * This directive binds input masking via Text Mask. Once the element is
+ * destroyed it will be unbinded. Pass an options object, same as you would
+ * normally for this plugin, as a parameter on the element being bound.
+ */
+
+let textMaskInputElement
+
+// Use core function to init the mask
+const setupTextMask = options => {
+  textMaskInputElement = createTextMaskInputElement(options)
+}
+
+// Register the input handler to watch for updates
+const inputHandler = ({target: {value}}) => {
+  return textMaskInputElement.update(value)
+}
+
+export default {
+  params: ['maskOptions'],
+  /**
+   * via Vue API
+   *
+   * Use the element (should be a text input) calling the directive as an
+   * option for the text mask. Expect options as params, but if none are sent
+   * just use the element as the only option. Send all options together in an
+   * object.
+   */
+  bind() {
+    // Get options as params
+    let options = this.params.maskOptions || {}
+    // Set the input as an option for text-mask
+    options.inputElement = this.el
+    setupTextMask(options)
+    this.el.addEventListener('input', inputHandler)
```

Comment:
I'm having a hard time with this one. The `update()` method you can use with Vue directives, does not fire when the model for the element is changed, it's for reacting to data change in some other way. Typically this is where you just add an event listener in JS, but we can't because you want < IE9 support. I did my best to find a good solution for adding event listeners to input text fields, but could not find anything that A) does not require jQuery, B) come with quite a bit of code. Here's how `v-model` works on [text input](https://github.com/vuejs/vue/blob/dev/src/directives/public/model/text.js). I'm stuck for the moment. I will update you if I get anything working.